### PR TITLE
First version of refactored clinical data export.

### PR DESCRIPTION
**Pull request by forus**
Refactored version of functionality is not equals to old one. It misses:
- Sample column
- Export of MRNA samples columns: tissue, timepoint,... (from assays table)
- Merging with SNP data

### CODE REVIEW

File: grails-app/services/com/recomdata/transmart/data/export/ClinicalExportService.groovy [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+        File clinicalDataFile = new File(studyDir, FILE_NAME)
+        clinicalDataFile.withWriter { Writer writer ->
+            CSVWriter csvWriter = new CSVWriter(writer, TAB)
+            int rowNumber = 1
+            for(def row: data) {
+                if (jobIsCancelled(args.jobName)) {
+                    return
+                }
+                if (rowNumber == 1) {
```
Comment (cataphract) [inline grails-app/services/com/recomdata/transmart/data/export/ClinicalExportService.groovy]: Should be moved out of the loop. Use a peeking iterator if necessary.

File: grails-app/services/com/recomdata/transmart/data/export/ClinicalExportService.groovy [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                csvWriter.writeNext(rowList as String[])
+
+                rowNumber += 1
+            }
+        }
+
+        clinicalDataFile
+    }
+
+    def boolean jobIsCancelled(jobName) {
```
Comment (cataphract) [inline grails-app/services/com/recomdata/transmart/data/export/ClinicalExportService.groovy]: `def` is redudant.

File: grails-app/services/com/recomdata/transmart/data/export/ClinicalExportService.groovy [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        if (jobResultsService[jobName]["Status"] == "Cancelled") {
+            log.warn("${jobName} has been cancelled")
+            return true
```
Comment (cataphract) [inline grails-app/services/com/recomdata/transmart/data/export/ClinicalExportService.groovy]: return is redundant.

File: grails-app/services/com/recomdata/transmart/data/export/ClinicalExportService.groovy [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        }
+        return false
```
Comment (cataphract) [inline grails-app/services/com/recomdata/transmart/data/export/ClinicalExportService.groovy]: idem

**Comment by forus** [comment]
@cataphract Please do not merge yet. There is one thing I'm going to add to this PR later: more test assertions for the exported file content.

**Comment by forus** [comment]
@cataphract Done. There are test assertions to check file content.

**Pull request merged**